src/OrchardCore.Modules/OrchardCore.LetsEncrypt/Controllers/AdminController.cs

```diff
@@ -0,0 +1,77 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Localization;
+using Microsoft.Extensions.Options;
+using OrchardCore.DisplayManagement.Notify;
+using OrchardCore.LetsEncrypt.Services;
+using OrchardCore.LetsEncrypt.Settings;
+using OrchardCore.LetsEncrypt.ViewModels;
+
+namespace OrchardCore.LetsEncrypt.Controllers
+{
+    public class AdminController : Controller
+    {
+        private readonly IAzureWebAppService _azureWebAppService;
+        private readonly ICertificateManager _certificateManager;
+        private readonly ILetsEncryptService _letsEncryptService;
+        private readonly LetsEncryptAzureAuthSettings _azureAuthSettings;
+        private readonly INotifier _notifier;
+
+        public AdminController(
+            IAzureWebAppService azureWebAppService,
+            ICertificateManager certificateManager,
+            ILetsEncryptService letsEncryptService,
+            IOptions<LetsEncryptAzureAuthSettings> azureAuthSettings,
+            INotifier notifier,
+            IHtmlLocalizer<AdminController> localizer
+            )
+        {
+            _azureWebAppService = azureWebAppService;
+            _certificateManager = certificateManager;
+            _letsEncryptService = letsEncryptService;
+            _azureAuthSettings = azureAuthSettings.Value;
+            _notifier = notifier;
+
+            T = localizer;
+        }
+
+        public IHtmlLocalizer T { get; }
+
+        public async Task<IActionResult> ManageCertificates()
+        {
+            return View(await BuildManageCertificatesViewModel());
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> InstallCertificate(InstallCertificateViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                await BuildManageCertificatesViewModel();
+                return View(model);
```

Comment:
This is not working because it requests a View named InstallCertificate.cshtml which I don't think is necessary here. It should just properly use the !ModelState.IsValid and return an error message back to the form stating that the Installation did not properly work because you forgot to set/select the proper values in the form.